### PR TITLE
[transformation][exec] allow spaces in parameters by enclosing in single quotes

### LINKED ISSUE #10158: [exec] Spaces in escaped argument result in multiple arguments
I just bumped openHAB from v2.5.0 to v3.0.1 and I am experiencing an issue with the exec binding. I hope that I just didn't miss anything in my update process but it looks to me like a bug with the exec add-on.

## My setup

I use the exec binding for a transformation inside a MQTT channel with the following setup.
```
Type dimmer : brightness "Brightness" [ stateTopic="zigbee2mqtt/Hue_Filament_Standard_E27_1/set", commandTopic="zigbee2mqtt/Hue_Filament_Standard_E27_1/set", transformationPatternOut="EXEC:/usr/local/bin/zlt brightness out %s", transformationPattern="EXEC:/usr/local/bin/zlt brightness in '%s'", retained=true]
```

Note the single quotes surrounding the  `%s` argument of the transformation pattern. A common expanded string for `s` is `{ "state": "ON/OFF", "brightness": 0-255 }`.

## Expected Behavior
In openHAB v2.5.0, using the single quotes around the `%s` argument resulted in the expanded string being considered one argument, not multiple space-separated ones. In my case, this resulted in an argument list of the exec script that looks like this:

Argument No. | Content
--- | ---
0 | /usr/local/bin/zlt
1 | brightness
2 | in
3 | { "state": "ON/OFF", "brightness": 0-255 }

## Current Behavior
With the update to v3.0.1, the single quotes no longer seem to be respected and result in individual arguments, broken on every space character like this:

Argument No. | Content
--- | ---
0 | `/usr/local/bin/zlt`
1 | `brightness`
2 | `in`
3 | `'{`
4 | `"state":`
5 | `"ON/OFF",`
6 | `"brightness":`
7 | ` 0-255`
8 | `}'`

## Steps to Reproduce (for Bugs)
1. Call an exec binding's script with a parameter that contains spaces but is enclosed by single quotes (e.g. { "state": "ON/OFF", "brightness": 0-255 }`
2. Print all the passed arguments in the script, e.g. using:
```
#!/bin/bash 
for i; do 
   echo $i
done
```

## Context
As said in the preamble, this issue only appeared after upgrading to openHAB v3.0.1. The setup as described above worked without issues in the previous v2.5.0 release.

## Your Environment
* openHAB: v3.0.1
* Exec binding: v3.0.1
* OS: Linux raspberrypi 5.4.83-v7l+ #1379 SMP Mon Dec 14 13:11:54 GMT 2020 armv7l GNU/Linux

### PR DESCRIPTION
**Pull request by lsiepel**
- Fix some readme issues
 - Fixes #10158

Credits to @J-N-K

### CODE REVIEW

**Comment by lolodomo** [comment]
@lsiepel : is it a breaking change that can make existing setups no more working ?

**Comment by lsiepel** [comment]
> @lsiepel : is it a breaking change that can make existing setups no more working ?

I cannot fully garantee it won't break any setup, but i can't think of a situation that it would actually break. If so i would also rather call it a bug.

File: bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java [hunk 1]
```diff
@@ -35,6 +36,7 @@
 @NonNullByDefault
 @Component(property = { "openhab.transform=EXEC" })
 public class ExecTransformationService implements TransformationService {
+    private static final Pattern SPLIT_ON_SPACE = Pattern.compile("(['])((?:\\\\\\1|.)+?)\\1|([^\\s']+)");
```
Comment (lolodomo) [inline bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java:39]: Are you really sure you need 6 consecutive backslash ?

Comment (lsiepel) [reply bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java:39]: The solution from @J-N-K was copied directly. I mainly looked at the single quotes without line breaks. 
And that looked oke. I cannot tell if there is a specific usecase that needs these backslashes. Maybe @J-N-K  can comment.

Comment (lolodomo) [reply bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java:39]: @J-N-K : please comment, I would like to merge this PR

Comment (lolodomo) [reply bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java:39]: @lsiepel : Can you point me to the original code from @J-N-K ?
We have to conclude here ;)

Comment (lsiepel) [reply bundles/org.openhab.transform.exec/src/main/java/org/openhab/transform/exec/internal/ExecTransformationService.java:39]: https://github.com/J-N-K/openhab-addons/pull/22/files

**Review by lolodomo — APPROVED** [review]
Let's merge and see if there are wrong feedback.

**Pull request merged**